server.c

```diff
@@ -19,32 +21,44 @@ struct request {
     char query[2048];
 };
 
-void request_parse(struct request* req, const char* data) {
+#define RESPONSE_TEMPLATE_MACRO "HTTP/1.1 200 OK\r\n"\
+        "Server: yeet/1.0\r\n"\
+        "Content-Type: text/plain\r\n\r\n"
+/* We don't want to actually save the null byte */
+char RESPONSE_TEMPLATE[sizeof(RESPONSE_TEMPLATE_MACRO) - 1] = RESPONSE_TEMPLATE_MACRO;
+
+#define MIN(one, two) (((one) < (two)) ? (one) : (two))
+
+int request_parse(struct request* req, const char* data) {
     // Make a copy of the request data for internal use
     char idata[65535];
-    strcpy(idata, data);
+    strncpy(idata, data, sizeof(idata) - 1);
+    idata[sizeof(idata) - 1] = 0;
```

Comment:
```suggestion
    idata[sizeof(idata) - 1] = '\0';
```
Looks lik idata is a `char*`, and while `0` isn't wrong, I'd prefer the explicit null byte char.